### ISSUE TITLE
TypeError: Cannot assign to read only property 'exports' of object '#<Object>'

### ISSUE THREAD

**Issue by durgaprasadksv** [react-native-community/react-native-device-info#950]
# Bug
When importing react-native-device-info in a web project (react-native-web) I get the following error. 

TypeError: Cannot assign to read only property 'exports' of object '#<Object>'
Module.<anonymous>
/Users/skamuju/hq/packages/web/index.js:115
  112 | /**
  113 |  * react-native-web empty polyfill.
  114 |  */
> 115 | module.exports = {
  116 |   getInstallReferrer: async () => {
  117 |     return getInstallReferrerSync();
  118 |   },

Project created using create-react-app v3
Library version: 5.5.1
React : 16.12.0

**Comment by mikehardy** [comment]
Hi there! I must admit I do not use this module for the web, and while there is a PR to add an example (you might compare with that?) there aren't automated tests to make sure it works either I don't think. There might be a problem or it might be an integration issue, I'm not sure. Perhaps @robertherber could lend a hand (he's been filing web PRs)

**Comment by robertherber** [comment]
@durgaprasadksv Could you check if the [example page](https://robertherber.github.io/react-native-device-info/) works in your environment? And if so does [this branch](https://github.com/robertherber/react-native-device-info/tree/example-web) - which is waiting to be merged - solve it for you?

**Comment by wong2** [comment]
same here

**Comment by wong2** [comment]
maybe related: https://github.com/webpack/webpack/issues/4039

**Comment by PZenha** [comment]
Facing same issue. Installed example-web but import "react-native-device-info" says "Can't find module".

**Comment by egoossens** [comment]
Same error on React Native Web

**Comment by mikehardy** [comment]
We need a new maintainer here that actively uses this on react-native-web. Or multiple really. There are a few PRs stacked up for web but I don't use it, so I won't maintain it well. Any takers? Just reviewing / testing those PRs and merging them might be all that's needed here, otherwise someone here needs to propose fresh PRs that fix it

**Comment by jamesst20** [comment]
Same error. Took me a while before figuring out it was caused by this library.

**Comment by mikehardy** [comment]
@jamesst20 very sorry for the trouble - I'm just not using react-native-web so it doesn't get great support, and I don't have time to learn about and bring up that toolchain. We need more maintainers :sweat:

**Comment by jamesst20** [comment]
@mikehardy Yeah I understand that. 

It's most likely due to mixing module.exports with import.

```
Quoting: https://github.com/webpack/webpack/issues/4039

You can mix require and export. You can't mix import and module.exports.
```

I'm pretty sure you would be able to refactor the 'module.exports' to something better :)

**Comment by mikehardy** [comment]
@jamesst20 It's astonishing how many tasks I have higher priority then this unfortunately. PRs gratefully, happily accepted

**Comment by jamesst20** [comment]
> @jamesst20 It's astonishing how many tasks I have higher priority then this unfortunately. PRs gratefully, happily accepted

Yeah. I will see if I can take a look. I can see you already have a few PRs awaiting since a while, will you get time to merge mine if I were to?

**Comment by mikehardy** [comment]
Yes - I'd give it priority, I really do want react-native-web to be supported, what I'm missing is someone that can really test things and say "good" (or "not good") so I don't merge something that blows people up. I don't want to make things worse, in other words - but I'm blind on react-native-web testing. Anything you said worked, I'd merge

**Comment by jamesst20** [comment]
@mikehardy Here you go :)

https://github.com/react-native-community/react-native-device-info/issues/950

**Comment by mikehardy** [comment]
I think this was merged and fixed now - closing but obviously reopen if I am mistaken
